keyboards/matrix/pjf/pjf.c

```diff
@@ -4,6 +4,51 @@
 
 #include "pjf.h"
 
+#ifdef RGBLIGHT_ENABLE
+#include "ws2812.h"
+#include "rgblight.h"
+
+extern rgblight_config_t rgblight_config;
+static bool pjf_led_mode = true;
+
+void rgblight_set(void) {
+    if (!rgblight_config.enable) {
+        for (uint8_t i = 0; i < RGBLED_NUM; i++) {
+            led[i].r = 0;
+            led[i].g = 0;
+            led[i].b = 0;
+        }
+    } else if( pjf_led_mode) {
+        led_t led_state = (led_t)host_keyboard_leds();
+        if( !led_state.num_lock) {
+            led[0].r = 0;
+            led[0].g = 0;
+            led[0].b = 0;
+        }
+        if( !led_state.caps_lock) {
+            led[1].r = 0;
+            led[1].g = 0;
+            led[1].b = 0;
+        }
+    }
+    ws2812_setleds(led, RGBLED_NUM);
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    switch(keycode) {
+    case KC_F24:
```

Comment:
Replacing `KC_F24` isnt really ideal, and if required, should use its own keycode.

Reply:
while using qmk configurator, no custom keycode can be used.

Reply:
> while using qmk configurator, no custom keycode can be used.

This isn't entirely true.

If you define the keycode logic at the keyboard level, QMK Configurator can use it. Satisfaction75 does this.

- [keycode logic](https://github.com/qmk/qmk_firmware/blob/2d051d8de3787de889ff0f0301153dbb4e2e1272/keyboards/cannonkeys/satisfaction75/satisfaction75.c#L186-L232)
- [result](https://config.qmk.fm/#/cannonkeys/satisfaction75/rev1/LAYOUT_default) – all the keycodes defined in the link above work in QMK Configurator.

Reply:
Satisfaction 75 defined the following key codes:
```
enum my_keycodes {
  ENC_PRESS = 0x5F80,
  CLOCK_SET,
  OLED_TOGG
};
```
While using the configurator's ANY key, it used ENSS, so how to map this code to the real custom keycode? thanks.

Reply:
The whole keycode is there, the HTML input element is just too small to show it all.

![image](https://user-images.githubusercontent.com/18669334/72197444-4745dc00-33d6-11ea-898d-76e5bf562a53.png)

As for the `0x5F80`, the Satisfaction75 runs [VIA Configurator](https://caniusevia.com/), which [defines a pre-set range for custom keycodes](https://github.com/qmk/qmk_firmware/blob/2d051d8de3787de889ff0f0301153dbb4e2e1272/quantum/via.h#L110-L127)

If you're not configuring for VIA support, you would use `SAFE_RANGE` in place of `0x5F80`, and then end with `NEW_SAFE_RANGE`:

```
enum my_keycodes {
  ENC_PRESS = SAFE_RANGE,
  CLOCK_SET,
  OLED_TOGG,
  NEW_SAFE_RANGE
};
```

... and then the keymaps need to use `NEW_SAFE_RANGE` in place of `SAFE_RANGE`. [MxSS does this](https://github.com/qmk/qmk_firmware/blob/2d051d8de3787de889ff0f0301153dbb4e2e1272/keyboards/mxss/keymaps/default/keymap.c#L31-L33) (though the keycode in that keymap has no logic; it's an example).

Reply:
thanks, current an custom keycode was introduced to switch the led mode